src/Compilers/CSharp/Test/Symbol/Symbols/Source/BaseClassTests.cs

```diff
@@ -1902,5 +1902,177 @@ class D : B {
             Assert.Equal(SpecialType.System_Int32, typeInfo.Type.SpecialType);
             Assert.Equal(SpecialType.System_Int64, typeInfo.ConvertedType.SpecialType);
         }
+
+        [Fact, WorkItem(5697, "https://github.com/dotnet/roslyn/issues/5697")]
+        public void InheritThroughStaticImportOfGenericTypeWithConstraint_01()
+        {
+            var text =
+@"
+using static CrashTest.Crash<CrashTest.Class2>; 
+
+namespace CrashTest 
+{ 
+    class Class2 : AbstractClass 
+    { 
+    } 
+
+    public static class Crash<T> 
+        where T: Crash<T>.AbstractClass 
+    { 
+        public abstract class AbstractClass 
+        { 
+            public int Id { get; set; } 
+        } 
+    } 
+}";
+            var comp = CreateCompilationWithMscorlib(text);
+            CompileAndVerify(comp);
+        }
+
+        [Fact, WorkItem(5697, "https://github.com/dotnet/roslyn/issues/5697")]
+        public void InheritThroughStaticImportOfGenericTypeWithConstraint_02()
+        {
+            var text =
+@"
+using static CrashTest.Crash<object>; 
+
+namespace CrashTest 
+{ 
+    class Class2 : AbstractClass 
+    { 
+    } 
+
+    public static class Crash<T> 
+        where T: Crash<T>.AbstractClass 
+    { 
+        public abstract class AbstractClass 
+        { 
+            public int Id { get; set; } 
+        } 
+    } 
+
+    class Class3
+    {
+        AbstractClass Test()
+        {
+            return null;
+        }
+    }
+}";
+            var comp = CreateCompilationWithMscorlib(text);
+            comp.VerifyDiagnostics(
+    // (6,11): error CS0311: The type 'object' cannot be used as type parameter 'T' in the generic type or method 'Crash<T>'. There is no implicit reference conversion from 'object' to 'CrashTest.Crash<object>.AbstractClass'.
```

Comment:
Why is this error being reported on line 6?  This seems to be the error generated by the following using which is on line 2. 

```
using static CrashTest.Crash<CrashTest.Class2>; 
```